src/cmap/connection_pool.ts

```diff
@@ -872,17 +886,19 @@ export class ConnectionPool extends TypedEventEmitter<ConnectionPoolEvents> {
           }
         } else {
           if (err) {
-            this.emit(
-              ConnectionPool.CONNECTION_CHECK_OUT_FAILED,
-              // TODO(NODE-5192): Remove this cast
-              new ConnectionCheckOutFailedEvent(this, 'connectionError', err as MongoError)
+            // TODO(NODE-5192): Remove this cast
+            const connectionCheckOutFailedEvent = new ConnectionCheckOutFailedEvent(
+              this,
+              'connectionError',
+              err as MongoError
             );
+            this.emit(ConnectionPool.CONNECTION_CHECK_OUT_FAILED, connectionCheckOutFailedEvent);
+            this.client.mongoLogger.debug('connection', connectionCheckOutFailedEvent);
           } else if (connection) {
             this[kCheckedOut].add(connection);
-            this.emit(
-              ConnectionPool.CONNECTION_CHECKED_OUT,
-              new ConnectionCheckedOutEvent(this, connection)
-            );
+            const connectionCheckedOutEvent = new ConnectionCheckedOutEvent(this, connection);
+            this.emit(ConnectionPool.CONNECTION_CHECKED_OUT, connectionCheckedOutEvent);
```

Comment:
Intead of editing each emit line would we want to consider an override? Perhaps there are events where we don't log? Just an idea.

```
  override emit<EventKey extends keyof ConnectionPoolEvents>(
    event: symbol | EventKey,
    ...args: Parameters<ConnectionPoolEvents[EventKey]>
  ): boolean {
    const res = super.emit(event, ...args);
    this.client.mongoLogger.debug('connection', args[0]);
    return res;
  }
```

Reply:
I would imagine that if we were to introduce any new events to the CMAP spec, we would also want those to be logged as well.

The only issue I have with this is that it doesn't allow for logging at a different severity level, which I think may be useful. In which case, I think the first argument should be the severity.

Reply:
From slack, lets gather emitting and logging into a shared method, it can always use debug, and be defined on TypedEE, internally